Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -129,5 +129,32 @@ def dumpcask
         odebug "Cask instance method '#{method}':", send(method).to_yaml
       end
     end
+
+    def to_hash
+      hsh = {
+        "name" => name,
+        "homepage" => homepage,
+        "url" => url,
+        "appcast" => appcast,
+        "version" => version,
+        "sha256" => sha256,
+        "artifacts" => {},
+        "caveats" => caveats,
+        "depends_on" => depends_on,
+        "conflicts_with" => conflicts_with,
+        "container" => container,
+        "gpg" => gpg,
+        "accessibility_access" => accessibility_access,
+        "auto_updates" => auto_updates,
+      }
+
+      artifacts.each do |a|
+        hsh["artifacts"][a.class.english_name] = a.summarize
```

Comment:
This should use `dsl_key` instead of `english_name`. And I think `artifacts` should be an array, not a hash. Also, this should call `to_h` or `to_a` on the artifact, not `summarize`.

Reply:
Re: call `to_h` or `to_a` on the artifact

I'm not sure how this should work, with different types of artifact having different methods. Please see my latest commit, using `methods.include?`. Is that how it should be?